src/app/tasks/components/TaskLoggerItem/TaskLoggerItem.tsx

```diff
@@ -72,17 +73,19 @@ const TaskLoggerItem = ({ notification }: TaskLoggerItemProps): JSX.Element => {
         onMouseLeave={handleMouseLeave}
       >
         <notification.icon className="h-8 w-8 drop-shadow-sm" />
-        <div className="flex flex-1 flex-col overflow-hidden text-left" title={notification.title}>
-          <span className="truncate text-sm font-medium text-gray-80">{notification.title}</span>
-
+        <div className="flex flex-1 flex-col overflow-hidden text-left">
+          <span className="truncate text-sm font-medium text-gray-80" title={notification.title}>
+            {notification.title}
+          </span>
           <span className={`text-sm ${messageClassName}`}>{notification.subtitle}</span>
         </div>
         <TaskLoggerActions
+          taskId={notification.taskId}
           isHovered={isHovered}
           status={notification.status}
           progress={progress.toString()}
           cancelAction={onCancelButtonClicked}
-          retryAction={isDownloadError ? retryDownload : () => {}}
+          retryAction={isDownloadError ? retryDownload : retryUpload}
```

Comment:
I would use a function here, as it is not clear why if it is not a download error, why we retry the upload